### ISSUE TITLE
html rendering doesn't change cross links .md to .html

### ISSUE THREAD

**Issue by metheoryt** [shd101wyy/markdown-preview-enhanced#781]
I need to render whole directory of .md files, which have cross links. And rendered .html files preserve cross-links on .md files.

**Comment by sheljohn** [comment]
There is a setting that you can change; in VSCode, this is the option `markdown-preview-enhanced.wikiLinkFileExtension`. Not sure what this looks like in Atom.

**Comment by houghtonap** [comment]
@sheljohn I don't believe that is what @metheoryt has an issue with. The issue, which I also have, is that when you have an .md file and it contains a link to another .md file, then you render it to HTML by right click the preview window and selecting HTML>HTML (offline), the links are rendered in the html to the .md file rather than to an .html file. For example x.md contains:

`[Technology Definitions](markup/glossary-technology.md)`

will be rendered in the .html file as:

`<a href="markup/glossary-technology.md">Technology Definitions</a>`

which isn't useful when you are trying to provide an HTML representation of the markdown. You would have to **manually** go into each HTML document and change the **.md** extension to an **.html** extension.

The other part, I believe, of @metheoryt issue is there is no way to render documents in a batch either by following links in the previewed document when you right click and select HTML>HTML (offline) or by some other means to specify the documents or directory you wish to convert.

Hope that helps clarify the issue for the Assigne or if I mis-characterized his issue, then Assigne split my similar issue out of this one.

**Comment by sheljohn** [comment]
@houghtonap [Wiki links](https://help.github.com/articles/adding-links-to-wikis/) are not normal links, the markup is `[[ text | path/to/file ]]` where the filepath is without extension.

---

Example folder:

```
.
├── a.md
└── b.md

0 directories, 2 files
```
with `a.md`:
```md
---
export_on_save:
  html: true
---

Link to [[b]].
```
and `b.md`:
```md
This is b.
```
with the option `"markdown-preview-enhanced.wikiLinkFileExtension": ".html"` should yield a file `a.html` with a link to `b.html`.
